gqlalchemy/vendors/memgraph.py

```diff
@@ -432,3 +442,13 @@ def with_power_bi(self) -> "Memgraph":
         module_name = "power_bi_stream.py"
 
         return self.add_query_module(file_path=file_path, module_name=module_name)
+
+    def get_storage_mode(self) -> str:
+        """Returns the storage mode of the Memgraph instance."""
+        result = self.execute_and_fetch("SHOW STORAGE INFO;")
+        storage_mode_value = next((item["value"] for item in result if item["storage info"] == "storage_mode"), None)
+        return MemgraphStorageMode(storage_mode_value).value
+
+    def set_storage_mode(self, storage_mode: MemgraphStorageMode) -> None:
+        """Sets the storage mode of the Memgraph instance."""
+        self.execute(f"STORAGE MODE {storage_mode};")
```

Comment:
Given that changing storage modes is not allowed while Memgraph contains data, could you write an integration test for that case and, if necessary, add exception handling here?

Reply:
Is it true that changing storage mode to on-disk from in-memory is not allowed, while switching from one in-memory storage mode to another is okay?

Reply:
You’re right @katarinasupe (https://memgraph.com/docs/fundamentals/storage-memory-usage#switch-storage-modes)

Reply:
I see we have an exception added in case there is data in the db and I try to switch to on disk storage mode:

> Client received query exception: You cannot switch from an in-memory storage mode to the on-disk storage mode when the database contains data. Delete all entries from the database, run FREE MEMORY and then repeat this query. 

I will add this exception handling and a test for that.